mgmtsystem/virtualcenter.py

```diff
@@ -131,58 +127,142 @@ def __init__(self, hostname, username, password, **kwargs):
         self.hostname = hostname
         self.username = username
         self.password = password
-        self._api = None
+        self._service_instance = None
+        self._content = None
         self._vm_cache = {}
         self.kwargs = kwargs
 
+    def __del__(self):
+        """Disconnect from the API when the object is deleted"""
+        if self._service_instance:
+            Disconnect(self._service_instance)
```

Comment:
i would strongly suggest to use a context manager, __del__ is not called deterministic ally and may cause issues at interpreter shutdown